[Merged by Bors] - feat: the `sinc` function

RemyDegenne opened the pull request:
- Define the `sinc` function, with value `sinc x = sin x / x` for `x ≠ 0` and `1` for `x = 0`
- Prove that `sinc` is continuous
- Add lemmas about measurability and integrability of `sinc`

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

urkud commented:
Do you have derivatives available in this file? A nice way to prove continuity of `sinc` is to say that it's `dslope sin 0`. BTW, do you plan to prove that `sinc` is analytic?

j-loreaux commented:
Yury stole all my questions. :laughing:

RemyDegenne commented:
> Do you have derivatives available in this file? A nice way to prove continuity of `sinc` is to say that it's `dslope sin 0`. BTW, do you plan to prove that `sinc` is analytic?

With the current imports, `deriv` exists but not `dslope`.
I don't plan to prove anything else about `sinc`. My only use for that function right now is #24846, in which I use it merely as notation for the if-then-else, and don't use any other property than the simple bounds I proved here (and integrability). I was not sure whether I should actually introduce a definition of `sinc` just for that, but since it's an otherwise useful function I figured it could not hurt.
I don't even need the continuity, but I thought it would be nice to add it (in this PR I use it to get measurability, but it could be replaced by another one-liner). I won't prove that it's analytic.

I could remove the continuity proof if you think it should be done in a different way.

j-loreaux commented on Mathlib/Analysis/SpecialFunctions/Trigonometric/Sinc.lean:
```diff
@@ -0,0 +1,89 @@
+
+variable {x : ℝ}
+
+/-- The function `sin x / x` mofified to take the value 1 at 0, which makes it continuous. -/
```
```suggestion
/-- The function `sin x / x` modified to take the value 1 at 0, which makes it continuous. -/
```

j-loreaux commented on Mathlib/MeasureTheory/Function/SpecialFunctions/Sinc.lean:
```diff
@@ -0,0 +1,60 @@
+
+## Main statements
+
+* `measurable_sinc`: the sinc function is measurable.
+* `integrable_sinc`: the sinc function is integrable with respect to any finite measure on `ℝ`.
+
```
Maybe you can add some TODOs for the Taylor series representation, analyticity, etc.

j-loreaux reviewed (approved):
I think it's fine to leave the other properties as TODOs.

bors d+

eric-wieser commented on Mathlib/Analysis/SpecialFunctions/Trigonometric/Sinc.lean:
```diff
@@ -0,0 +1,89 @@
+
+## Main definitions
+
+* `Real.sinc`: the sinc function, defined as `sinc x = sin x / x` for `x ≠ 0` and `1` for `x = 0`.
```
```suggestion
* `Real.sinc`: the (unnormalized) sinc function, defined as `sinc x = sin x / x` for `x ≠ 0` and `1` for `x = 0`.
```

since https://en.wikipedia.org/wiki/Sinc_function suggests that different fields use a different scale factor. Perhaps worth adding the link to the references, even if its obvious.

eric-wieser commented on Mathlib/Analysis/SpecialFunctions/Trigonometric/Sinc.lean:
```diff
@@ -0,0 +1,89 @@
+
+/-- The function `sinc` is continuous. -/
+@[fun_prop]
+lemma continuous_sinc : Continuous sinc := by
```
Could you add a todo to show it is also differentiable, and even ContDiff?

eric-wieser replied:
(or perhaps a `proof_wanted`!)

RemyDegenne commented:
I am not a fan of adding TODOs. Nearly 100% of the TODOs I added in Mathlib outlived their implementation (mostly because I myself forgot to remove them once I implemented them), which results in misleading todos in the file docstrings.
Also anyone who wants to use this function in an application that needs more than its mere definition will naturally prove differentiability, analycity, etc.
So unless someone insists again on having TODOs in the next few hours, I'll merge without them.

eric-wieser commented:
Are `proof_wanted`s better than TODOs in that regard?

j-loreaux commented:
I'm fine if you leave out the TODOs

RemyDegenne commented:
bors r+
